spec/models/student_spec.rb

```diff
@@ -2,4 +2,25 @@
 
 RSpec.describe Student, "validations" do
   it { is_expected.to validate_presence_of(:name) }
+  it { is_expected.to validate_presence_of(:grade) }
+  it { is_expected.to validate_presence_of(:teacher_id) }
+  it { is_expected.to validate_presence_of(:hair_color)}
 end
+
+RSpec.describe Student, type: :model do
+  describe '#change_hair' do
+    it 'should change hair_color to black' do
+      student = Student.create(hair_color: 'red')
+      student.change_hair
+      expect(student.hair_color).to eq('green')
+    end
+  end
+end
```

Comment:
Well done! This is a great test, it's simple enough to understand what is going on. 👍